### PR TITLE
feat(MenuItem): Add all label capabilities to menu item

### PR DESCRIPTION
**Pull request by rivka-ungar**
https://monday.monday.com/boards/3532714909/views/80492480/pulses/7077366107

### CODE REVIEW

File: packages/core/src/components/Menu/MenuItem/MenuItem.tsx [hunk 1]
```diff
@@ -116,6 +116,16 @@ const MenuItem: VibeComponent<MenuItemProps | MenuItemTitleComponentProps> & {
...
 
+    const renderLabel = useCallback(() => {
+      if (!label) return;
+      if (typeof label === "string") {
+        return <Label kind={Label.kinds.LINE} text={label} />;
+      }
+      if (React.isValidElement(label) && label.type === Label) {
+        return label;
+      }
+    }, [label]);
```
Comment (YossiSaadi) [inline packages/core/src/components/Menu/MenuItem/MenuItem.tsx:127]: I think `useMemo` can be a better candidate here
and you can render it afterwards without calling the function, just as {renderLabel}

File: packages/core/src/components/Menu/MenuItem/MenuItem.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ import { SubmenuPosition } from "./MenuItem.Types";
 
 export interface MenuItemProps extends VibeComponentProps {
   title?: string;
-  label?: string;
+  label?: string | React.ReactElement<typeof Label>;
```
Comment (YossiSaadi) [inline packages/core/src/components/Menu/MenuItem/MenuItem.tsx:20]: that's a great solution! I initially thought about adding a labelProps, but that's way better if we already want to allow everything!

File: packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js [hunk 3]
```diff
@@ -67,6 +68,26 @@ describe("<MenuItem />", () => {
...
+    await waitFor(() => expect(labelElement).toBeTruthy());
+  });
+
+  it("should render the Label component with props when pass a component", async () => {
+    const labelText = "Label Text";
+    const { getByTestId } = render(
+      <MenuItem title={title} label={<Label text={labelText} color="dark" kind="line" />} />
+    );
+    const labelElement = getByTestId("label");
+    await waitFor(() => expect(labelElement).toBeTruthy());
```
Comment (YossiSaadi) [inline packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js:84]: is `waitFor` needed here?

File: packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js [hunk 4]
```diff
@@ -67,6 +68,26 @@ describe("<MenuItem />", () => {
...
+    const { getAllByTestId } = within(labelElement);
+    const labelTextElement = getAllByTestId("text")[0];
```
Comment (YossiSaadi) [inline packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js:86]: does it have more than one test id of text? if so, we should take a look into it, sound like a bug

**Review by YossiSaadi — COMMENTED** [review]
had one suggestion.
great solution!

File: packages/core/src/components/Menu/MenuItem/MenuItem.tsx (hunk 1 above)
Comment (rivka-ungar) [reply packages/core/src/components/Menu/MenuItem/MenuItem.tsx]: You're right, I changed it

File: packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js (hunk 3 above)
Comment (rivka-ungar) [reply packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js]: Nope, removed!

**Review by YossiSaadi — APPROVED** [review]
Awesome job!

File: packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js (hunk 4 above)
Comment (YossiSaadi) [reply packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js:86]: (not in this PR of course)

Comment (rivka-ungar) [reply packages/core/src/components/Menu/MenuItem/__tests__/MenuItem.test.js:86]: Yes, In the label there are 2 nested text elements, not sure why maybe for the animation?

**Pull request merged**